django-backend/fecfiler/web_services/summary/summary.py

```diff
@@ -47,6 +59,29 @@ def calculate_summary(self):
         summary["line_33"] = summary["line_11d"]
         return summary
 
+    def calculate_summary_column_b(self):
+        committee = self.report.committee_account
+        report_date = self.report.coverage_through_date
+        report_year = report_date.year
+
+        ytd_transactions = Transaction.objects.filter(
+            committee_account=committee,
+            date__year=report_year,
+            date__lt=report_date,
+        )
+
+        # line 15
+        sa15_query = Q(~Q(memo_code=True), form_type="SA15")
+        line_15 = self._create_contribution_sum(sa15_query)
+
+        # build summary
+        summary = ytd_transactions.aggregate(
+            line_15=line_15,
+            line_37=line_15
```

Comment:
sorry i didn't clarify this well.  You'll want to assign line 37 outside the query.  As it is, the ORM could still make the line_15 subquery twice, once for line_15 and once for line_37.  At the end of `calculate_summary_column_a()` the lines based on other lines are calculated  after the query, so that there's no double work.

Reply:
:+1: